oslc_prolog/lib/oslc_core.pl

```diff
@@ -101,20 +101,22 @@
     once((
       resource_md5(IRI, Graph, ReceivedHash),
       copy_resource(IRI, IRI, rdf(GraphIn), rdf(Graph), []),
-      format('Status: 204~n~n')
-    ; format('Status: 412~nContent-type: text/plain~n~nThe value of "If-Match" header does not match resource [~w]~n', [IRI])
+      throw(status_code(204))
+    ; rdf_global_id(LIRI, IRI),
+      format(atom(Message), 'The value of "If-Match" header does not match resource [~w]', [LIRI]),
+      throw(status_code(412, Message))
     ))
   ),
     oslc_error(Message),
-    format('Status: 400~nContent-type: text/plain~n~n~w~n', [Message])
+    throw(status_code(400, Message))
   ).
 
 handle_delete(_, IRI) :-
   catch((
     autodetect_resource_graph(IRI, Graph),
     delete_resource(IRI, rdf(Graph)),
-    format('Status: 200~n~n')
+    throw(status_code(200))
```

Comment:
204? https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete